mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -289,6 +289,33 @@ impl mayastor_server::Mayastor for MayastorSvc {
         .await
     }
 
+    #[instrument(level = "debug", err)]
+    async fn set_nvme_ana_state(
+        &self,
+        request: Request<SetNvmeAnaStateRequest>,
+    ) -> GrpcResult<Null> {
+        let args = request.into_inner();
+        let uuid = args.uuid.clone();
+        debug!("Setting NVMe ANA state for nexus {} ...", uuid);
+
+        let ana_state = match NvmeAnaState::from_i32(args.ana_state) {
```

Comment:
This is not wrong or anything but, have you looked at the `matches!()` macro? IMHO it makes the code directly more obvious and comes by default with the system.

Reply:
what about using the trait `TryFrom<i32>` ?

Reply:
Indeed, `std::convert::TryFrom<i32> for NvmeAnaState` seems reasonable.

Reply:
AIUI the `TryFrom<i32>` has to be in the same crate as `::rpc::mayastor::NvmeAnaState`, otherwise I get "compiler says no". Anyway, this is exactly the same pattern used for `ShareProtocolNexus` several lines earlier so probably `matches!()` would be suitable.